client/src/features/ProjectPageV2/ProjectPageContent/Documentation/Documentation.tsx

```diff
@@ -0,0 +1,277 @@
+  toggle: () => void;
+}
+
+function DocumentationModal({
+  isOpen,
+  project,
+  toggle,
+}: DocumentationModalProps) {
+  const [updateProject, result] = usePatchProjectsByProjectIdMutation();
+  const { isSuccess, isLoading, error } = result;
+
+  const {
+    control,
+    formState: { errors },
+    handleSubmit,
+    getValues,
+    register,
+    reset,
+    setValue,
+    watch,
+  } = useForm<DocumentationForm>({
+    defaultValues: {
+      documentation: project.documentation || "",
+    },
+  });
+
+  useEffect(() => {
+    setValue("documentation", project.documentation || "");
+  }, [project.documentation, setValue]);
+
+  const onSubmit = useCallback(
+    (data: DocumentationForm) => {
+      updateProject({
+        "If-Match": project.etag ? project.etag : "",
+        projectId: project.id,
+        projectPatch: { documentation: data.documentation },
+      });
+    },
+    [project.etag, project.id, updateProject]
+  );
+
+  useEffect(() => {
+    if (!isOpen) {
+      reset({ documentation: project.documentation || "" });
+      result.reset();
+    }
+  }, [isOpen, project.documentation, reset, result]);
+
+  useEffect(() => {
+    if (result.isSuccess) {
+      toggle();
+    }
+  }, [result.isSuccess, toggle]);
+
+  const documentationField = register("documentation", {
+    maxLength: {
+      message: `Documentation is limited to ${DESCRIPTION_MAX_LENGTH} characters.`,
+      value: DESCRIPTION_MAX_LENGTH,
+    },
+  });
+  return (
+    <ScrollableModal
+      backdrop="static"
+      centered
+      data-cy="project-documentation-modal"
+      isOpen={isOpen}
+      size="lg"
+      toggle={toggle}
+    >
+      <ModalHeader toggle={toggle} data-cy="project-documentation-modal-header">
+        <div>
+          <FileEarmarkText className={cx("me-1", "bi")} />
+          Documentation
+        </div>
+      </ModalHeader>
+      <Form noValidate onSubmit={handleSubmit(onSubmit)}>
+        <ModalBody
+          data-cy="project-documentation-modal-body"
+          className={styles.modalBody}
+        >
+          <div className="mb-1">
+            <TextAreaInput<DocumentationForm>
+              control={control}
+              getValue={() => getValues("documentation")}
+              name="documentation"
+              register={documentationField}
+            />
+          </div>
+        </ModalBody>
+        <ModalFooter
+          className="border-top"
+          data-cy="project-documentation-modal-footer"
+        >
+          {errors.documentation ? (
+            <div className="text-danger">
+              {errors.documentation.message ? (
+                <>{errors.documentation.message}</>
+              ) : (
+                <>Documentation text is invalid</>
+              )}
+            </div>
+          ) : (
+            <div className="mb-2"></div>
+          )}
+          {isSuccess != null && !isSuccess && (
+            <RtkOrNotebooksError error={error} />
+          )}
```

Comment:
`result.error` will only be populated in case of error from the API.

```suggestion
          {result.error && (
            <RtkOrNotebooksError error={result.error} />
          )}
```